modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -596,6 +643,212 @@ private QueryPlan prepareDml(SqlNode sqlNode, PlanningContext ctx) throws Valida
         return new MultiStepDmlPlan(template, queryFieldsMetadata(ctx, igniteRel.getRowType(), null));
     }
 
+    /** */
+    private QueryPlan prepareDdl(SqlNode sqlNode, PlanningContext ctx) {
+        assert sqlNode instanceof SqlDdl : sqlNode == null ? "null" : sqlNode.getClass().getName();
+
+        SqlDdl ddlNode = (SqlDdl)sqlNode;
+
+        if (ddlNode instanceof IgniteSqlCreateTable) {
+            IgniteSqlCreateTable createTblNode = (IgniteSqlCreateTable)ddlNode;
+
+            SqlIdentifier tblFullName = createTblNode.name();
+
+            CreateTableCommand createTblCmd = new CreateTableCommand();
+
+            createTblCmd.schemaName("PUBLIC"); // TODO: get from the context
+            createTblCmd.tableName(Util.last(tblFullName.names));
+            createTblCmd.ifNotExists(createTblNode.ifNotExists());
+            createTblCmd.templateName(QueryUtils.TEMPLATE_PARTITIONED);
+
+            if (createTblNode.createOptionList() != null) {
+                for (SqlNode optNode : createTblNode.createOptionList().getList()) {
+                    IgniteSqlCreateTableOption opt = (IgniteSqlCreateTableOption)optNode;
+
+                    switch (opt.key()) {
+                        case TEMPLATE:
+                            if (!(opt.value() instanceof SqlIdentifier))
+                                throw new IgniteSQLException("Unexpected value for param " + TEMPLATE + " [" +
+                                    "expected an identifier, but was " + opt.value() + "; " +
+                                    "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.PARSING);
+
+                            createTblCmd.templateName(((SqlIdentifier)opt.value()).names.get(0));
+                            break;
+
+                        case BACKUPS:
+                            if (!(opt.value() instanceof SqlNumericLiteral)
+                                || !((SqlNumericLiteral)opt.value()).isInteger()
+                                || ((SqlLiteral)opt.value()).intValue(true) < 0
+                            )
+                                throw new IgniteSQLException("Unexpected value for param " + BACKUPS + " [" +
+                                    "expected a non-negative integer, but was " + opt.value() + "; " +
+                                    "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.PARSING);
```

Comment:
we have a few places for throwing parsing exception, maybe better to extract a separate method for it and pass only exception text and context?

Reply:
fixed